daft/io/lance/lance_data_sink.py

```diff
@@ -91,7 +92,7 @@ def write(self, micropartitions: Iterator[MicroPartition]) -> Iterator[WriteResu
         lance = self._import_lance()
 
         for micropartition in micropartitions:
-            arrow_table = micropartition.to_arrow()
+            arrow_table = pa.Table.from_batches(micropartition.to_arrow().to_batches(), self._pyarrow_schema)
```

Comment:
I believe the problem with this PR is this change may be disregarding the daft-specific schema information. In the roundtrip embeddings tests, we're specifically checking to make sure that we're serializing & deserializing fixed size arrays of a specific numeric type. We expect these values to always be accepted as / be returned as `numpy` arrays. (**Not** python `list`s).

The failing test's error on this PR is saying that the schema that's saved doesn't align to the one we expected to be saved. @ddupg can you please try to inspect why this is changing the schema information when saved in Daft? What's the difference between what was being saved beforehand and what's being saved now?

Reply:
Thanks @malcolmgreaves so much for taking the time to look into this PR and share your suggestions! I truly appreciate the detailed guidance you’ve provided.